docs/upgrading.rst

```diff
@@ -0,0 +1,447 @@
+`use_args` call, and `"json"` is the default. If `q` is intended to be a query
+parameter, you must be explicit and rewrite like so:
+
+.. code-block:: python
+
+    # webargs 5.x
+    @parser.use_args({"q": ma.fields.String()})
+    def foo(q):
+        return some_function(user_query=q)
+
+
+    # webargs 6.x
+    @parser.use_args({"q": ma.fields.String()}, location="query")
+    def foo(q):
+        return some_function(user_query=q)
+
+This also means that another usage from 5.x is not supported. If you had code
+with multiple locations in a single `use_args`, `use_kwargs`, or `parse` call,
+you must write it in multiple separate `use_args` invocations, like so:
+
+.. code-block:: python
+
+    # webargs 5.x
+    @parser.use_args(
+        {
+            "q1": ma.fields.Int(location="query"),
+            "q2": ma.fields.Int(location="query"),
+            "h1": ma.fields.Int(location="headers"),
+        },
+        locations=("query", "headers"),
+    )
+    def foo(q1, q2, h1):
+        ...
+
+
+    # webargs 6.x
+    @parser.use_args({"q1": ma.fields.Int(), "q2": ma.fields.Int()}, location="query")
+    @parser.use_args({"h1": ma.fields.Int()}, location="headers")
+    def foo(q1, q2, h1):
+        ...
+
+
+Fields no longer support location=...
+-------------------------------------
+
+Because a single `parser.use_args`, `parser.use_kwargs`, or `parser.parse` call
+cannot specify multiple locations, it is not necessary for a field to be able
+to specify its location. Rewrite code like so:
+
+.. code-block:: python
+
+    # webargs 5.x
+    @parser.use_args({"q": ma.fields.String(location="query")})
+    def foo(q):
+        return some_function(user_query=q)
+
+
+    # webargs 6.x
+    @parser.use_args({"q": ma.fields.String()}, location="query")
+    def foo(q):
+        return some_function(user_query=q)
+
+location_handler has been replaced with location_loader
+-------------------------------------------------------
+
+This is not just a name change. The expected signature of a `location_loader`
+is slightly different from the signature for a `location_handler`.
+
+Where previously your code took the incoming request data and details of a
+single field being loaded, it now takes the request and the schema as a pair.
+It does not return a specific field's data, but data for the whole location.
+Rewrite code like this:
+
+.. code-block:: python
+
+    # webargs 5.x
+    @parser.location_handler("data")
+    def load_data(request, name, field):
+        return request.data.get(name)
+
+
+    # webargs 6.x
+    @parser.location_loader("data")
+    def load_data(request, schema):
+        return request.data
+
+Data is not filtered before being passed to your schema, and it may be proxified
+--------------------------------------------------------------------------------
+
+In webargs 5.x, the schema you gave was used to pull data out of the request
+object. That data was compiled into a dictionary which was then passed to your
+schema.
```

Comment:
In webargs 5.x, the deserialization schema was used to pull data out of the request
object. That data was compiled into a dictionary which was then passed to the schema.